### ISSUE TITLE
Database fails to save under C++ API

### ISSUE THREAD

**Issue by robert-yates** [Vector35/binaryninja-api#726]
Binary Ninja Version: 1.0.794-dev, a22d60e6
Platform: Windows 10

Build the API and edit the bin-info sample from the examples folder, then add logging
`LogToFile(BNLogLevel::DebugLog, "c:\\users\\public\\log.txt", false); `
and after the analysis try to save a database
```
        std::string output = "C:\\Users\\Public\\dbTest\\Test.exe.bndb";
	if (!bv->CreateDatabase(output))
		printf("Error saving  database %08X\n", GetLastError());
```

`Microsoft C++ exception: BinaryNinjaCore::DatabaseException`
and in the log `Error initializing database: Unable to read file contents to store into database.`

The same happens under linux, i tried using different path formats etc.

GetLastError() reports `ERROR_HANDLE_EOF` additionally a 12kb truncated db file
is created always.

[Test.zip](https://github.com/Vector35/binaryninja-api/files/1124787/Test.zip)

**Comment by psifertex** [comment]
What line number are you adding that code to? It's unfortunately a bit of a pain to troubleshoot since we use a different build system internally that doesn't build those plugins, but I the API itself does work -- it's the same API that the UI calls to save a database.

**Comment by robert-yates** [comment]
Here is an example of what i used under linux, i marked the 2 lines with `// TEST`, i did also notice that python api works from within the gui if i type `bv.create_database("/tmp/test.bndb")` however not from a compiled cpp program. something very odd is going on 😄 

[bin-info.zip](https://github.com/Vector35/binaryninja-api/files/1125861/bin-info.zip)

**Comment by robert-yates** [comment]
The easiest way to reproduce/debug this would be on ubuntu,
simply install the latest dev branch of binja or eventually your
debug build then:

`
cd ~
https://github.com/Vector35/binaryninja-api.git
cd binaryninja-api/
make
cd examples/bin-info/
`
copy the Test.exe to ~ and the bin-info.cpp over the one
in the examples/bin-info/ then:
`
cp ~/binaryninja/libbinaryninjacore.so.1 ~/binaryninja/libbinaryninjacore.so
make
bin/bininfo ~/Test.exe
`

on another note, i dont know why you have to copy and rename the core lib
for `make` to work

regards,
Rob.

**Comment by plafosse** [comment]
My guess is you aren't calling `InitCorePlugins()` which is required for initialization.

**Comment by robert-yates** [comment]
I just checked my test code and i am calling InitCorePlugins() and i double checked the path is correct

**Comment by robert-yates** [comment]
Just to keep this issue up to date, the latest information is that:
bv->ReadBuffer() returns always a 0 length DataBuffer

**Comment by plafosse** [comment]
Just looked into this again.

One way to fix this could have been to do:
```
if (!bv->GetParentView()->CreateDatabase("/tmp/test.bndb"))  // TEST
```
I've fixed the API so it does this for you automatically so your original code should work. Sorry about the delay in getting this fixed.
Fixed in build 1.1.1126
